tests/codeception/functional/Install/Content/VideoEmbedCest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * Class VideoEmbedCest.
+ *
+ * @group install
+ */
+class VideoEmbedCest {
+
+  /**
+   * I can create a page with a video embed and verify caption.
+   */
+  public function testVideoEmbed(FunctionalTester $I) {
+    // Login and add flexible page
+    $I->logInWithRole('contributor');
+    $I->amOnPage('node/add/hs_basic_page');
+    $I->fillField('Title', 'Demo Basic Page');
+
+    // Add text field
+    $I->scrollTo('#edit-field-hs-page-components-add-more');
+    $I->click('List additional actions', '#edit-field-hs-page-components-add-more');
+    $I->scrollTo('#field-hs-page-components-hs-text-area-add-more');
+    $I->click('#field-hs-page-components-hs-text-area-add-more');
+    $I->wait(2);
+
+    // Add media
+    $I->click('.cke_button__drupalmedialibrary');
+    $I->waitForText('Add or select media');
+    $I->click('.media-library-menu__link[data-title="Video"]');
+    $I->waitForText('Add Video via URL');
+    $I->fillField('Add Video via URL', 'https://www.youtube.com/watch?v=95N_spFNEkY');
+    $I->click('.js-form-submit[value="Add"]');
+    $I->executeJS('document.querySelector("body").removeChild(document.querySelector(".ui-widget-overlay"))', []);
+    $I->waitForText('Save and select');
+    $I->executeJS('document.querySelector(".ui-dialog-buttonpane .button--primary").click()', []);
+    $I->waitForText('Insert selected');
+    $I->executeJS('document.querySelector("button.media-library-select").click()', []);
```

Comment:
Why are you using javascript to do anything here? you shouldn't click or remove anything via javascript. A non-technical user can't do this, so you shouldn't be doing it.

Reply:
Removed JS and replaced with clicking button text with CSS classes to refine.